packages/babel-generator/src/generators/classes.js

```diff
@@ -10,10 +10,8 @@ export function ClassDeclaration(node: Object, parent: Object) {
 
   this.word("class");
 
-  if (node.id) {
-    this.space();
-    this.print(node.id, node);
-  }
+  this.space();
+  this.print(node.id, node);
```

Comment:
These should stay. The generator should be able to serialize anything our AST can represent, there is no guarantee that someone is using it alongside `babel-traverse` and will have used the scoping APIs.